source/PlayerInfo.cpp

```diff
@@ -1842,7 +1845,7 @@ void PlayerInfo::Visit(const System *system)
 // Mark the given planet as visited.
 void PlayerInfo::Visit(const Planet *planet)
 {
-	if(planet && !planet->TrueName().empty())
+	if(planet)
 		visitedPlanets.insert(planet);
```

Comment:
The proper fix is to ensure (at load-time) that TrueName is never empty. Still, if we do encounter a corrupted savegame that causes TrueName to be empty, then I prefer to leave it unvisited.